src/sensor/ping.cpp

```diff
@@ -241,15 +243,17 @@ void Ping::handleMessage(PingMessage msg)
 
         // This is necessary to convert <uint8_t> to <int>
         // QProperty only supports vector<int>, otherwise, we could use memcpy
-        for (int i = 0; i < m.num_points(); i++) {
-            _points.replace(i, m.data()[i] / 255.0); // TODO we should really be working in ints
+        qCDebug(PING_PROTOCOL_PING) << "length:" << m.profile_data_length();
+
+        for (int i = 0; i < m.profile_data_length(); i++) {
+            _points.replace(i, m.profile_data()[i] / 255.0); // TODO we should really be working in ints
```

Comment:
Why we should work with ints ?
The idea to use floats is to normalize the data from the sensor, 0 being the lowest and 1 the highest output that the sensor can return. This makes everything else after this point abstract for filters, widgets and etc.

Reply:
it's ok. Maybe I have been working without fpu too often.

would it be worth the effort/performance to normalize to 0~UINT16MAX?

Reply:
there is just a lot of high frequency math operations done against float from message->display that could be done in int math, that was my idea.